src/main/java/com/fasterxml/jackson/databind/node/NodeCursor.java

```diff
@@ -195,7 +195,10 @@ public JsonToken nextToken()
         public ObjectCursor(JsonNode n, NodeCursor p)
         {
             super(JsonStreamContext.TYPE_OBJECT, p);
-            _contents = ((ObjectNode) n).fields();
+            if (!n.isObject()) {
```

Comment:
It is debatable is this check is really needed, `ArrayCursor` for example has no such check.

Reply:
Was about to say that: please just remove this check, not needed.